wgengine/userspace.go

```diff
@@ -314,6 +321,31 @@ func echoRespondToAll(p *packet.ParsedPacket, t *tstun.TUN) filter.Response {
 	return filter.Accept
 }
 
+// handleLocalPackets inspects packets coming from the local network
+// stack, and intercepts any packets that should be handled by
+// tailscaled directly. Other packets are allowed to proceed into the
+// main ACL filter.
+func (e *userspaceEngine) handleLocalPackets(p *packet.ParsedPacket, t *tstun.TUN) filter.Response {
+	if e.useTailscaleDNS {
+		if verdict := e.handleDNS(p, t); verdict == filter.Drop {
+			// local DNS handled the packet.
+			return filter.Drop
+		}
+	}
+
+	// TODO: atomic value
+	localAddrs := e.localAddrs.Load().(map[packet.IP]bool)
+	if localAddrs[p.DstIP] {
+		// Packet is destined for the local system, but got
+		// misdirected into the tun interface (probably we're on
```

Comment:
should we restrict this to macOS for performance reasons for now until needed elsewhere?